pkg/hubble/relay/observer/server.go

```diff
@@ -168,6 +169,65 @@ sortedFlowsLoop:
 	return g.Wait()
 }
 
+// GetNodes implements observerpb.ObserverClient.GetNodes.
+func (s *Server) GetNodes(ctx context.Context, req *observerpb.GetNodesRequest) (*observerpb.GetNodesResponse, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		ctx = metadata.NewOutgoingContext(ctx, md)
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
+
+	peers := s.peers.List()
+	nodes := make([]*observerpb.Node, 0, len(peers))
+	for _, p := range peers {
+		n := &observerpb.Node{
+			Name: p.Name,
+			Tls: &observerpb.TLS{
+				Enabled:    p.TLSEnabled,
+				ServerName: p.TLSServerName,
+			},
+		}
+		if p.Address != nil {
+			n.Address = p.Address.String()
+		}
+		nodes = append(nodes, n)
+		if !isAvailable(p.Conn) {
+			n.State = relaypb.NodeState_NODE_UNAVAILABLE
+			s.opts.log.WithField("address", p.Address).Infof(
+				"No connection to peer %s, skipping", p.Name,
+			)
+			s.peers.ReportOffline(p.Name)
+			continue
+		}
+		n.State = relaypb.NodeState_NODE_CONNECTED
+		p := p
```

Comment:
🙄

Reply:
@Rolinh perhaps would be good to add a comment to explain the motivation?

Reply:
@errordeveloper This is common practice and even recommended as the "easier" way of achieving this [in the Go FAQ](https://golang.org/doc/faq#closures_and_goroutines). I could add a comment here but then there are many places in the codebase that would need the same comment.

Reply:
ah, if it's actually part of the FAQ, comment is unnecessary